src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.Accessor.cs

```diff
@@ -246,7 +246,13 @@ private bool WriteStream(TKey key, Stream stream, Checksum? checksum, Cancellati
 
                         return t.connection.ReadDataBlob_MustRunInTransaction(t.database, t.self.Table, t.rowId);
                     },
-                    (self: this, connection, database, checksum, rowId));
+                    (self: this, connection, database, checksum, rowId),
+                    throwOnSqlException: true);
+
+                // we should never have gotten a SqlException while reading since we passed throwOnSqlException: true above.
+                Contract.ThrowIfTrue(exception != null);
```

Comment:
Isn't there a `Contract.ThrowIfNotNull`?

Reply:
there is not :)